### ISSUE TITLE
Get `:nonode@nohost` in 'config-and-releases' chapter

### ISSUE THREAD

**Issue by cj1128** [elixir-lang/elixir-lang.github.com#1391]
I am following the official tutorial of building a simple key-value store. 

According to this chapter https://elixir-lang.org/getting-started/mix-otp/config-and-releases.html, I add this in `config.exs`

```elixir
config :kv, :routing_table, [{?a..?z, node()}]
```

There is no problem in development. But when I create a release and start it `./_build/prod/rel/foo/bin/foo start_iex` I will get an error saying `no connection to nonode@nohost`.

I checked the `:routing_table` config and it's `:nonode@nohost`.

```elixir
iex(foo@computer-name)1> Application.get_env(:kv, :routing_table)
iex(foo@computer-name)2> [{97..122, :"nonode@nohost"}]
```

I tried to put the same content in `releases.exs` but it doesn't work either.

I guess the reason is that when elixir runs the config file, it uses a process that doesn't have name set so `node()` would return `:nonode@nohost`?

I can avoid this problem by removing the call of `node()` in `config.exs` and using a static config instead.

But I want to know what is the correct way of fixing this problem? Say I want to make the routing table have only one entry pointing to whatever the current node is.

PS: It would be much helpful if there is an official repo for the kv app in the tutorial. Anyway, thank you guys for all the hard work, the current documentation is really readable and helpful.

**Comment by josevalim** [comment]
Hi @cj1128! Putting it in config/releases.exs should work. What is the value of `node()` inside config/releases.exs? Otherwise, the config is set when the code is compiled, which is not what you want.

Btw, the example app can be found here: https://github.com/josevalim/kv_umbrella

**Comment by cj1128** [comment]
Thank you for the quick response @josevalim 

Actually it doesn't work, the value is still `:nonode@nohost`. I guess although the code is running at runtime, the process used to run it still isn't alive?

The repo is helpful. I can't believe I forgot to search in your repo!

**Comment by josevalim** [comment]
@cj1128 oh, there is a flag called `start_distributed` in your release configuration that you have to enable. Otherwise the app is not running distributed during configuration. Although it may be better to set this value when you start the app then, instead of using a config file...

**Comment by josevalim** [comment]
Or you can use the RELEASE_NAME or RELEASE_NODE environment variables to build it by hand.

**Comment by cj1128** [comment]
@josevalim I believe what you are saying is `:start_distribution_during_config` according to [Mix.Tasks.Release](https://hexdocs.pm/mix/Mix.Tasks.Release.html#module-config-providers).

I confirmed that if I set it true for the foo release config in `mix.exs`, the error would be gone.

I know setting the routing table in the config file is a bad idea. I just want to make the kv_umbrella in the tutorial work. 😄 

Thank you for the timely help!
